omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java

```diff
@@ -140,19 +156,20 @@ private void BuildJSON() {
 			log.error("Error while creating Swagger specification", e);
 		}
 		finally {
-			toggleLogs(RestConstants.SWAGGER_LOGS_ON);
+			toggleLogs(SwaggerConstants.SWAGGER_LOGS_ON);
 		}
 	}
 
 	public String getJSON() {
-		if (isCached()) {
+		if (isCached() && cachedJson != null) {
 			log.info("Returning a cached copy of Swagger specification");
-			initOpenAPI();
-		} else {
-			openAPI = new OpenAPI();
-			BuildJSON();
+			return cachedJson;
 		}
-		return createJSON();
+
+		openAPI = new OpenAPI();
+		BuildJSON();
+		cachedJson = createJSON();  // Cache the JSON string
```

Comment:
The Spec was being reset when accessing cached data, causing
empty paths in the API documentation on subsequent requests. This occurred
because `initOpenAPI()` was creating a new empty Paths object while handling
cached responses.